helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/AvroCompatibilityHelper.java

```diff
@@ -281,11 +281,23 @@ public static Decoder newCompatibleJsonDecoder(Schema schema, String in) throws
    * @return a decoder
    * @throws IOException on io errors
    */
-  public static Decoder newCachedResolvingDecoder(Schema writer, Schema reader, Decoder in) throws IOException {
+  public static SkipDecoder newCachedResolvingDecoder(Schema writer, Schema reader, Decoder in) throws IOException {
     assertAvroAvailable();
     return ADAPTER.newCachedResolvingDecoder(writer, reader, in);
   }
 
+  /**
+   * {@link Decoder} that fixes a bug in the BinaryDecoder that can cause OutOfMemoryError
```

Comment:
i think a better name is BoundedMemoryDecoder? i  assume all this does is protects against OOMs  when reading garbage?